### PR TITLE
feat(account): Add a way to add a new credit card to account

### PR DESCRIPTION
**Pull request by tadzik**

### CODE REVIEW

File: src/app/account-app/credit-cards/credit-cards.component.ts [hunk 1]
```diff
@@ -76,9 +79,26 @@ export class CreditCardsComponent implements OnInit {
...
+                dialogRef.afterClosed().subscribe(cardAdded => {
+                    if (cardAdded) {
+                        // not pretty, but forces the progress spinner to show up again,
+                        // and it's better than looking like nothing happened
+                        this.creditCards = new ReplaySubject<any>(1);
+                        this.refreshCards();
+                    }
+                });
+            },
+            _err => this.snackbar.open('Failed to add a new credit card', 'Okay'),
```
Comment (castaway) [inline src/app/account-app/credit-cards/credit-cards.component.ts:95]: What sort of error is this one? Can we get any reasons why it errored?

File: src/app/account-app/credit-cards/credit-cards.component.ts [hunk 2]
```diff
@@ -76,9 +79,26 @@ export class CreditCardsComponent implements OnInit {
...
+        this.rmmapi.addNewCard().subscribe(
+            res => {
+                const dialogRef = this.dialog.open(StripeAddCardDialogComponent, { data: { clientSecret: res.client_secret } });
+                dialogRef.afterClosed().subscribe(cardAdded => {
+                    if (cardAdded) {
+                        // not pretty, but forces the progress spinner to show up again,
+                        // and it's better than looking like nothing happened
+                        this.creditCards = new ReplaySubject<any>(1);
+                        this.refreshCards();
+                    }
```
Comment (castaway) [inline src/app/account-app/credit-cards/credit-cards.component.ts:92]: What about if !CardAdded? error snackbar? do we get any reasoning why it didnt?

File: src/app/account-app/credit-cards/credit-cards.component.ts (hunk 1 above)
Comment (castaway) [reply src/app/account-app/credit-cards/credit-cards.component.ts:95]: Mebbe use the "contact support" one? (tho unless support get some sort of error log, not all that helpful either)

File: src/app/account-app/credit-cards/stripe-add-card-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            <div class="payment-field payment-field-small">
+                <div #cardCvc></div>
+                <label for="cardCvc">CVC code</label>
+            </div>
+            <div class="payment-field payment-field-small StripeElement">
+                <mat-form-field>
+                    <input matInput [(ngModel)]="zipCode" placeholder="ZIP code">
+                </mat-form-field>
+            </div>
+        </div>
```
Comment (castaway) [inline src/app/account-app/credit-cards/stripe-add-card-dialog.component.html:69]: No "name" field?

**Review by castaway — COMMENTED** [review]
Few comments/queries

File: src/app/account-app/credit-cards/credit-cards.component.ts (hunk 1 above)
Comment (tadzik) [reply src/app/account-app/credit-cards/credit-cards.component.ts:95]: Yeah, we don't have a snackbar form of "contact support" yet – maybe this is a good time to introduce it, actually.

Comment (tadzik) [reply src/app/account-app/credit-cards/credit-cards.component.ts]: Fixed now in cde0ae33aff779f486420f4dfdc3fe3fb2fa0124

File: src/app/account-app/credit-cards/stripe-add-card-dialog.component.html (hunk 3 above)
Comment (tadzik) [reply src/app/account-app/credit-cards/stripe-add-card-dialog.component.html:69]: We don't need it for anything – should we have one anyway?

File: src/app/calendar-app/runbox-calendar-event.spec.ts [hunk 4]
```diff
@@ -204,7 +204,7 @@ describe('RunboxCalendarEvent', () => {
         expect(sut.toIcal()).toContain('DTSTART:20210201T100000');
     });
 
-     it('should be possible to add a special case to a recurring event (with timezone)', () => {
+    /*it('should be possible to add a special case to a recurring event (with timezone)', () => {
```
Comment (tadzik) [inline src/app/calendar-app/runbox-calendar-event.spec.ts:207]: Silly accidental commit

**Comment by gtandersen** [comment]
Perhaps the snakcbar could say "Error: Failed to add a new credit card." in white, and the the next line in light gray:
"Please try again later, or contact Runbox Support at support@runbox.com."

**Review by castaway — APPROVED** [review]
Decided not to rename "addNewCard" in rmmapi?  Looks good to me

**Pull request merged**